### PR TITLE
add get_val method

### PR DESCRIPTION
**Pull request by enricostragiotti**
This PR implement #565  
the old implementation suggested in #569 has been discarded

### CODE REVIEW

File: src/fastoad/openmdao/variables/variable.py [hunk 1]
```diff
@@ -135,6 +136,19 @@ def __init__(self, name, **kwargs):
         if not self.description and self.name in self._variable_descriptions:
             self.description = self._variable_descriptions[self.name]
 
+    def get_val(self, new_units: Union[str, None] = None):
+        """Method used to get the variable value converted in the new_units"""
```
Comment (christophe-david) [inline src/fastoad/openmdao/variables/variable.py:140]: Well, we already know it is a method 😉 

```suggestion
        """Returns the variable value converted in the `new_units`"""
```

**Review by christophe-david — CHANGES_REQUESTED** [review]
Thanks for the proposed feature.

It needs refinement, though. Just hoping I won't dishearten you 😅

File: src/fastoad/openmdao/variables/variable.py (hunk 1 above)
Comment (enricostragiotti) [reply src/fastoad/openmdao/variables/variable.py]: Modified in [b91d0ee](https://github.com/fast-aircraft-design/FAST-OAD/pull/570/commits/b91d0ee67ed3bb55ceabc5bbfa778fd9a5f7126f) 😅

File: src/fastoad/io/tests/test_data_file.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
 from ...openmdao.variables import Variable, VariableList
 
 RESULTS_FOLDER_PATH = Path(__file__).parent / "results" / Path(__file__).stem
+DATA_FOLDER_PATH = Path(__file__).parent / "data" / Path(__file__).stem
```
[the remark this answers is not in the document]
Comment (enricostragiotti) [reply src/fastoad/io/tests/test_data_file.py]: Modified in [b91d0ee](https://github.com/fast-aircraft-design/FAST-OAD/pull/570/commits/b91d0ee67ed3bb55ceabc5bbfa778fd9a5f7126f) 😅

File: src/fastoad/openmdao/tests/test_variables.py [hunk 3]
```diff
@@ -836,3 +837,22 @@ def test_get_variables_from_problem_sellar_with_promotion_and_connect():
...
+
+def test_get_val():
+    vars = VariableList()
+    vars["bar"] = {"value": 1.0, "units": "m"}
+    vars["foo"] = {"value": 1.0, "units": "m**2"}
+    vars["baz"] = {"value": [1.0, 2.0], "units": "m"}
+    vars["bat"] = {"value": (1.0, 2.0), "units": "m"}
+    data = vars["bar"].get_val()
+    assert_allclose(data, 1, rtol=1e-3, atol=1e-5)
+    units = "ft"
```
[the remark this answers is not in the document]
Comment (enricostragiotti) [reply src/fastoad/openmdao/tests/test_variables.py]: yes, true :)

File: src/fastoad/openmdao/variables/variable.py [hunk 4]
```diff
@@ -135,6 +136,19 @@ def __init__(self, name, **kwargs):
...
+        if new_units:
+            if isinstance(self.value, Iterable):
+                return [
+                    convert_units(self.value[i], self.units, new_units)
+                    for i in range(len(self.value))
+                ]
+            else:
+                return convert_units(self.value, self.units, new_units)
+        else:
+            return self.value
```
[the remark this answers is not in the document]
Comment (enricostragiotti) [reply src/fastoad/openmdao/variables/variable.py]: I like the idea of using either nparrays or float. I didn't know about scalarize :) thanks for the input!

File: src/fastoad/openmdao/variables/variable.py [hunk 5]
```diff
@@ -135,6 +136,19 @@ def __init__(self, name, **kwargs):
         if not self.description and self.name in self._variable_descriptions:
             self.description = self._variable_descriptions[self.name]
 
+    def get_val(self, new_units: Union[str, None] = None):
```
[the remark this answers is not in the document]
Comment (enricostragiotti) [reply src/fastoad/openmdao/variables/variable.py]: Modified in [b91d0ee](https://github.com/fast-aircraft-design/FAST-OAD/pull/570/commits/b91d0ee67ed3bb55ceabc5bbfa778fd9a5f7126f) 😅

**Review by christophe-david — APPROVED** [review]
All good!
Thank you!

**Pull request merged**